text/0113-canonical-metadata.md

```diff
@@ -0,0 +1,625 @@
+        * Class
+        * Properties
+        * Public methods
+    * Programmatically generated events
+        * type
+        * options: composed, bubbles
+    * Programmatically fetched static resources
+        * `<script>`
+        * `<img>`
+        * `<link>`
+* CSS file
+    * Tokens
+
+## Metadata shape
+
+```json
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "description": "A representation of metadata gathered for a LWC bundle",
+    "type": "object",
+    "properties": {
+        "name": { "type": "string" },
+        "namespace": { "type": "string" },
+        "templates": { 
+            "type": "array",
+            "items": { "$ref": "#/definitions/HTMLTemplate"}
+        },
+        "js": {
+            "type": "array",
+            "items": { "$ref": "#/definitions/JSFile"}
+        },
+        "defaultComponentClass": {
+            "type": "array",
+            "items": { 
+                "$ref": "#/definitions/Class"
+            }
+        },
+        "css": {
+            "$ref": "#/definitions/CSSFile"
+        }
+    },
+    "definitions": {
+        "HTMLTemplate": {
+            "type": "object",
+            "properties": {
+                "fileType": { "$ref": "#/definitions/FileType" },
+                "fileName": { "type": "string" },
+                "children": {
+                    "type": "array",
+                    "items": { 
+                        "$ref": "#/definitions/CustomElementReference"
+                    }
+                },
+                "staticResources": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/StaticResourceReference"
+                    }
+                },
+                "dynamicChildren": {
+                    "type": "array",
+                    "items": { 
+                        "$ref": "#/definitions/DynamicComponentReference"
+                    }
+                },
+                "directives": {
+                    "type": "object",
+                    "properties": {
+                        "forEach": {
+                            "type": "array", 
+                            "items": { "$ref" : "#/definitions/TemplateForEachDirective" }
+                        },
+                        "iterator": {
+                            "type": "array", 
+                            "items": { "$ref" : "#/definitions/TemplateIteratorDirective" }
+                        },
+                        "if": {
+                            "type": "array", 
+                            "items": { "$ref" : "#/definitions/TemplateIfDirective" }
+                        },
+                        "lwcDom": {
+                            "type": "array", 
+                            "items": { "$ref" : "#/definitions/TemplateLwcDomDirective" }
+                        }
+                    }
+                }
+            }
+        },
+        "JSFile": {
```

Comment:
Let's plan for a future where we might add support for Typescript. Let's convert this type to `Script`.

Reply:
🙀